quint/test/effects/inferrer.test.ts

```diff
@@ -1,39 +1,37 @@
 import { describe, it } from 'mocha'
 import { assert } from 'chai'
-import { LookupTable, LookupTableByModule, newTable } from '../../src/lookupTable'
+import { LookupTable, LookupTableByModule, mergeTables, newTable } from '../../src/lookupTable'
 import { buildModuleWithDefs } from '../builders/ir'
 import { effectSchemeToString } from '../../src/effects/printing'
 import { errorTreeToString } from '../../src/errorTree'
-import { defaultValueDefinitions } from '../../src/definitionsCollector'
-import { EffectInferrer } from '../../src/effects/inferrer'
+import { collectDefinitions, defaultValueDefinitions } from '../../src/definitionsCollector'
+import { EffectInferenceResult, EffectInferrer } from '../../src/effects/inferrer'
+import { QuintModule } from '../../src'
 
 describe('inferEffects', () => {
   const table: LookupTable = newTable({
     valueDefinitions: defaultValueDefinitions().concat([
-      { kind: 'param', identifier: 'p', reference: 1n },
-      { kind: 'const', identifier: 'N', reference: 2n },
-      { kind: 'var', identifier: 'x', reference: 3n },
-      { kind: 'val', identifier: 'm', reference: 2n },
-      { kind: 'val', identifier: 't', reference: 5n },
-      { kind: 'def', identifier: 'f', reference: 6n },
-      { kind: 'param', identifier: 'g', reference: 7n },
-      { kind: 'def', identifier: 'my_add', reference: 2n },
-      { kind: 'def', identifier: 'a', reference: 9n },
-      { kind: 'val', identifier: 'b', reference: 10n },
-      { kind: 'const', identifier: 'S', reference: 11n },
-      { kind: 'param', identifier: 'tup', reference: 12n },
+      { kind: 'const', identifier: 'N', reference: 1n },
+      { kind: 'const', identifier: 'S', reference: 1n },
+      { kind: 'var', identifier: 'x', reference: 1n },
     ]),
   })
 
-  const definitionsTable: LookupTableByModule = new Map<string, LookupTable>([['wrapper', table]])
+  function inferEffectsForModule(quintModule: QuintModule): EffectInferenceResult {
```

Comment:
I gave up on mocking definitions tables last week. This is an equivalent change to what I made on `modeChecker` tests recently: use the definitions collector instead of trying to keep these complicated mocks (as the id matters and it's super hard to figure out the correct ones).